enhancements/authentication/custom-scc-preemption-prevention.md

```diff
@@ -0,0 +1,249 @@
+- If the required SCC is not applicable, the workload fails to be scheduled.
+This can be detected in the existing e2e suite.
+- All core OpenShift workloads need to set that annotation in their deployment artifacts.
+This implies that, after the change merges, many repositories have to be touched.
+
+An existing implementation is available at https://github.com/openshift/apiserver-library-go/pull/90.
+
+### (Alternative) "soft protection" - Hinting a favored SCC
+
+Another suggested approach is to use hinting instead.
+Here, an annotation named `openshift.io/scc-hint` contains the hinted SCC name as its value.
+
+If the hinted SCC does not exist in the system, it is being ignored silently.
+In this case SCC admission continues with all remaining SCCs available in the system as described above.
+
+If the hinted SCC does exist in the system it is being prepended to the sorted list describe above,
+before the highest priority SCC.
+
+This ensures that the hinted SCC is deterministically considered to be applied first
+before any other SCC available in the system.
+
+Pros:
+- Less hard failure modes than the proposed "hard protection" mechanism.
+- Less intrusive than the "hard protection" alternative
+as the workload will continue to be asserted against the remaining SCCs available in the system.
+
+Cons:
+- If the hinted SCC is not applicable to the workload,
+admission continues with asserting all remaining SCCs available in the system as described above.
+- If the hinted SCC is not applicable, admission continues with asserting all the other SCCs available in the system,
+potentially even custom SCCs which still can cause problems.
+- All core OpenShift workloads need to set that annotation in their deployment artifacts.
+This implies that, after the change merges, many repositories have to be touched.
+- To verify that hinted SCCs are really being applied, an e2e test has to be written.
+This can be done centrally.
+
+An existing implementation is available at https://github.com/openshift/apiserver-library-go/pull/105.
+
+This proposal favors the "hard protection" variant and continues to describe that one further on.
+
+### Workflow Description
+
+Opting into this feature for core OpenShift workloads is straight-forward,
+with just an annotation needed to be set in the deployment manifest.
+
+### Variation (optional)
+
+Another variation for both proposed suggestions is to accept a comma-separated list of SCCs.
+This could be helpful in case additional SCCs are going to be introduced and the deployment manifest
+needs to be compatible with both old and new applicable SCCs.
+
+However, as adding a list further adds complexity, this suggestion is being omitted as an variation only
+with the potential of being introduced as a backwards-compatible change in the future.
+
+### API Extensions
+
+A new annotation `openshift.io/required-scc` is introduced.
+
+### Implementation Details/Notes/Constraints [optional]
+
+N/A
+
+### Risks and Mitigations
```

Comment:
Since the owners of the workloads are responsible for configuring the SCCs, do they have enough knowledge to make the right choices, or do they require support in selecting the appropriate ones?

As you mentioned, misconfigured SCCs should generally be detected by the e2e tests, right?

Reply:
i would assume yes